File: Demo/DJISdkDemo/DJISdkDemo/Demo/Navigation/HotPoint/HotPointConfigView.m
//
//  InspireHotPointConfigView.m
//  DJISdkDemo
//
//  Created by Ares on 15/6/3.
//  Copyright (c) 2015年 DJI. All rights reserved.
//

#import "HotPointConfigView.h"

@interface HotPointConfigView ()
@property (weak, nonatomic) IBOutlet UITextField *altitudeInputBox;
@property (weak, nonatomic) IBOutlet UITextField *radiusInputBox;
@property (weak, nonatomic) IBOutlet UITextField *speedInputBox;
@property (weak, nonatomic) IBOutlet UISegmentedControl *headingControl;
@property (weak, nonatomic) IBOutlet UISegmentedControl *entryControl;
@property (weak, nonatomic) IBOutlet UISwitch *clockwiseSwitch;

- (IBAction)onOkButtonClicked:(id)sender;

@end

@implementation HotPointConfigView

-(id) initWithNib
{
    UIView* mainView = [[[NSBundle mainBundle] loadNibNamed:@"HotPointConfigView" owner:self options:nil] objectAtIndex:0];
    self = [super initWithFrame:[mainView bounds]];
    if (self) {
        [self addSubview:mainView];
        
        _altitude = 50.0;
        _radius = 20.0;
        _speed = 20;
        _headingMode = DJIHotPointHeadingAlongTheCircleLookingForwards;
        _entryPoint = DJIHotPointEntryFromNorth;
        _clockwise = YES;
        
        self.altitudeInputBox.text = [NSString stringWithFormat:@"%0.1f", _altitude];
        self.radiusInputBox.text = [NSString stringWithFormat:@"%0.1f", _radius];
        self.speedInputBox.text = [NSString stringWithFormat:@"%d", _speed];
        [self.headingControl setSelectedSegmentIndex:(int)_headingMode];
        [self.entryControl setSelectedSegmentIndex:(int)_entryPoint];
        [self.clockwiseSwitch setOn:_clockwise];
    }
    
    return self;
}

-(void) setAltitude:(float)altitude
{
    self.altitudeInputBox.text = [NSString stringWithFormat:@"%0.1f", altitude];
}

- (IBAction)onOkButtonClicked:(id)sender {
    _altitude = [self.altitudeInputBox.text floatValue];
    _radius = [self.radiusInputBox.text floatValue];
    _speed = [self.speedInputBox.text intValue];
    _headingMode = (DJIHotPointHeadingMode)self.headingControl.selectedSegmentIndex;
    _entryPoint = (DJIHotPointEntryPoint)self.entryControl.selectedSegmentIndex;
    _clockwise = self.clockwiseSwitch.isOn;
    if (self.delegate && [self.delegate respondsToSelector:@selector(configViewWillDisappear)]) {
        [self.delegate configViewWillDisappear];
    }
    [UIView animateWithDuration:0.2 animations:^{
        self.alpha = 0.0;
    }];
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    if ([self.altitudeInputBox isFirstResponder]) {
        [self.altitudeInputBox resignFirstResponder];
    }
    if ([self.radiusInputBox isFirstResponder]) {
        [self.radiusInputBox resignFirstResponder];
    }
    if ([self.speedInputBox isFirstResponder]) {
        [self.speedInputBox resignFirstResponder];
    }
    return YES;
}
@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Missions/HotPoint/HotPointConfigView.swift
//
//  HotPointConfigView.swift
//  DJISDKSwiftDemo
//
//  Created by DJI on 15/6/3.
//  Copyright (c) 2015 DJI. All rights reserved.
//
import UIKit
import DJISDK
protocol HotPointConfigViewDelegate: UITextFieldDelegate {
    func configViewWillDisappear()
}
class HotPointConfigView: UIView {
    
    @IBOutlet weak var altitudeInputBox: UITextField!
    @IBOutlet weak var radiusInputBox: UITextField!
    @IBOutlet weak var speedInputBox: UITextField!
    @IBOutlet weak var headingControl: UISegmentedControl!
    @IBOutlet weak var entryControl: UISegmentedControl!
    @IBOutlet weak var clockwiseSwitch: UISwitch!
    
    weak var delegate: HotPointConfigViewDelegate? = nil

    
    var startPoint:DJIHotPointStartPoint {
        get {
            return DJIHotPointStartPoint(rawValue: UInt(self.entryControl.selectedSegmentIndex))!
        }
        
        set(startPoint) {
            self.entryControl.selectedSegmentIndex = Int(startPoint.rawValue)
        }
    
    }//=
    var heading:DJIHotPointHeading {
        get {
            return DJIHotPointHeading(rawValue: UInt(self.headingControl.selectedSegmentIndex))!
        }
        
        set(heading) {
            self.headingControl.selectedSegmentIndex = Int(heading.rawValue)
        }
    
    } //=
    
    var altitude: CGFloat {
        get {
            return CGFloat((self.altitudeInputBox.text! as NSString).floatValue)
        }
        set (altitude) {
            self.altitudeInputBox.text = String(format: "%0.1f", altitude)
        }
    }

    var radius: CGFloat {
        get {
            return  CGFloat((self.radiusInputBox.text! as NSString).floatValue)
        }
        
        set(radius) {
            self.radiusInputBox.text = String(format: "%0.1f", radius)
        }
    }

    var speed: Int {
        get {
            return Int((self.speedInputBox.text! as NSString).intValue)
        }
        
        set(speed) {
            self.speedInputBox.text = "\(speed)"
        }
    }

    var clockwise: Bool {
        get {
            return self.clockwiseSwitch.isOn
        }
        
        set (clockwise) {
            self.clockwiseSwitch.isOn = clockwise
        }
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    func initWithNib() -> HotPointConfigView {
        let mainView: UIView = Bundle.main.loadNibNamed("HotPointConfigView", owner: self, options: nil)![0] as! UIView
        self.frame = mainView.bounds
        self.addSubview(mainView)
        self.altitude = 50.0
        self.radius = 20.0
        self.speed = 20
        self.clockwise = true
        self.startPoint = DJIHotPointStartPoint.north
        self.heading = DJIHotPointHeading.alongCircleLookingForward
        return self
    }


    @IBAction func onOkButtonClicked(_ sender: AnyObject) {
      
        self.delegate?.configViewWillDisappear()
        UIView.animate(withDuration: 0.2, animations: {() -> Void in
            self.alpha = 0.0
        })
    }

    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        if self.altitudeInputBox.isFirstResponder {
            self.altitudeInputBox.resignFirstResponder()
        }
        if self.radiusInputBox.isFirstResponder {
            self.radiusInputBox.resignFirstResponder()
        }
        if self.speedInputBox.isFirstResponder {
            self.speedInputBox.resignFirstResponder()
        }
        return true
    }

}
